### DISCUSSION TITLE
Decryption Key Is Not Decrypted

### DISCUSSION THREAD

**Discussion by nschutz** [openpgpjs/openpgpjs#1437] [Q&A] [answered]
Hey I'm having an error message that I'm having trouble understanding. I apologize if this is the wrong place for this. 

The error I'm seeing is "Decryption Key is not decrypted" at opengpg.js line 29212. However if I try to use the openpgp.decryptKey function I get an error "Key packet is already decrypted".  It seems similar to this issue: https://github.com/openpgpjs/openpgpjs/issues/413 but comparing my code to the tests doesn't help, the private key only has '\n'  separating the lines. What could I be missing?

Here's the bits of code with blah replacing the key and passphrase:

```
      const privateKeyArmored = `-----BEGIN PGP PRIVATE KEY BLOCK-----

blah
-----END PGP PRIVATE KEY BLOCK-----
`;
      const encrypted = `-----BEGIN PGP MESSAGE-----
Version: PGP Command Line v10.4.1 (Build 54) (Linux)

blah
-----END PGP MESSAGE-----
`
      const passphrase = 'blah';

      const privateKey = await openpgp.decryptKey({
        privateKey: await openpgp.readPrivateKey({ armoredKey: privateKeyArmored }),
        passphrase
      });

      const message = await openpgp.readMessage({
          armoredMessage: encrypted // parse armored message
      });

      const { data: decrypted, signatures } = await openpgp.decrypt({
        message,
        decryptionKeys: privateKey
      });
```

**Comment by larabr** [comment, marked as answer]
Hi @nschutz , please try this solution: https://github.com/openpgpjs/openpgpjs/discussions/1431#discussioncomment-1519034

**Reply by nschutz** [reply]
It worked! Thank you so much I toiled with this for hours last night. Can I buy you a coffee on paypal or something?

**Reply by larabr** [reply]
Thanks, but no need :)
